### ISSUE TITLE
rt.sh is not working properly when -l and -n used together.

### ISSUE THREAD

**Issue by uturuncoglu** [ufs-community/ufs-weather-model#2244]
## Description
Today, I synced the `ufs-coastal` with `ufs-weather-model` head of develop. It seems that `rt.sh` script `-l` argument is not working anymore when it is combined with `-n` argument. This was working before with the following hash.

```
commit 5d2ca19d0a9ededdfd0199656c2e7fa096bf57a9 (origin/develop)
Author: WenMeng-NOAA <48260754+WenMeng-NOAA@users.noreply.github.com>
Date:   Fri Apr 19 13:59:12 2024 -0400
```

In this case, RT system is trying to run all the tests (I guess) starting from the first one from the specified file. This is very critical in the development since we are extensively using `-l` along with `-n` in the UFS coastal development to test specific configuration from the test config file specific for UFS coastal. I am not sure this is intentional or bug but it would be nice to preserve this feature.

### To Reproduce:

Hercules/Intel
1. checkout ufs-weather-model - git clone --recursive https://github.com/ufs-community/ufs-weather-model
2. cd ufs-weather-model/tests
3. and try to run a one test
    ./rt.sh -l rt.conf -a nems -k -n "datm_cdeps_lnd_gswp3 intel"

## Additional context
I know there is no meaning to use `rt.conf` since it is default but in UFS Coastal we are using `rt_coastal.conf`. So, this just a way of reproducing issue under `ufs-weather-model`.

## Output

Partly output of `rt.sh`. It is seen that it is trying to compile `s2swa_32bit_intel` rather than `datm_cdeps_land`.
```
+ cp fv3_conf/fv3_slurm.IN_hercules fv3_conf/fv3_slurm.IN
+ cp fv3_conf/compile_slurm.IN_hercules fv3_conf/compile_slurm.IN
+ eval 'set +x'
++ set +x
Linking /work2/noaa/stmp/tufuk/stmp/tufuk/FV3_RT/rt_2666619 to /work2/noaa/nems/tufuk/COASTAL/ufs-weather-model/tests/run_dir
Run regression test in: /work2/noaa/stmp/tufuk/stmp/tufuk/FV3_RT/rt_2666619
rt.sh: Checking & Updating test configuration...
No update needed to TESTS_FILE
rt.sh: Running compile s2swa_32bit_intel
```

**Comment by uturuncoglu** [comment]
FYI, @pvelissariou1 @janahaddad

**Comment by BrianCurtis-NOAA** [comment]
@uturuncoglu try this please: https://github.com/BrianCurtis-NOAA/ufs-weather-model/tree/fix_rtsh_bugs if it works I will get this into the develop branch. Thanks!

**Comment by uturuncoglu** [comment]
@BrianCurtis-NOAA Thanks. Sure. Let me test.

**Comment by uturuncoglu** [comment]
@BrianCurtis-NOAA Thanks for your help. I test and it seems it fixed the issue with the `rt.sh`. Thanks again.

**Comment by BrianCurtis-NOAA** [comment]
This was fixed in #2241
